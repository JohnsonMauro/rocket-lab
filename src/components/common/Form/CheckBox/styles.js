import styled from '@emotion/styled';
import Label from '@components/common/Form/Label';

const CheckBoxContainer = styled.div``;

const CheckBoxLabel = styled(Label)`
  cursor: pointer;
  display: flex;

  & input:checked ~ span:after {
    display: block;
  }
`;

const CheckBoxInput = styled.input`
  margin-left: 18px;
  opacity: 0;
  height: 0;
  width: 0;
`;

const CheckBoxSpan = styled.span`
  border: 0.6px solid ${({ theme }) => theme.colors.gainsboro};
  border-radius: 1.6px;
  height: 16px;
  left: 0;
  top: 0;
  width: 16px;

  &:after {
    content: '';
    border: solid ${({ theme }) => theme.colors.success};
    border-width: 0 1px 1px 0;
    display: none;
    height: 10px;
    left: 4px;
    position: relative;
    top: 1px;
    transform: rotate(45deg);
    width: 5px;
    -webkit-transform: rotate(45deg);
    -ms-transform: rotate(45deg);
    -webkit-transform: rotate(45deg);
    -ms-transform: rotate(45deg);
  }
`;

export { CheckBoxContainer, CheckBoxInput, CheckBoxLabel, CheckBoxSpan };
