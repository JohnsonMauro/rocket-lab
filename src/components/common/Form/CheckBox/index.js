import { CheckBoxInput, CheckBoxLabel, CheckBoxSpan } from './styles';

const CheckBox = ({ forControl, id, label, name, value }) => (
  <CheckBoxLabel forControl={forControl}>
    {label}
    <CheckBoxInput type="checkbox" id={id} name={name} value={value}></CheckBoxInput>
    <CheckBoxSpan></CheckBoxSpan>
  </CheckBoxLabel>
);

export default CheckBox;
