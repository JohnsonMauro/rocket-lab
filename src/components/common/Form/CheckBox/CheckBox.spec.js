import React from 'react';
import { render } from '@testing-library/react';

import GlobalsContainer from '@test/GlobalsContainer';
import CheckBox from './index';

describe('<CheckBox />', () => {
  it('Renders the checkBox component', () => {
    const { container } = render(
      <CheckBox forControl="checkTest" id="checkTest" label="test check box" name="checkTest" />,
      {
        wrapper: GlobalsContainer,
      }
    );
    expect(container).toMatchSnapshot();
  });
});
