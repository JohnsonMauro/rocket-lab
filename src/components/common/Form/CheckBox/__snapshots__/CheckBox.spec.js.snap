// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> Renders the checkBox component 1`] = `
<div>
  <label
    class="eynslpk1 css-fad30s-LabelStyle-CheckBoxLabel e6ushqh0"
    for="checkTest"
  >
    test check box
    <input
      class="css-i1akeh-CheckBoxInput eynslpk2"
      id="checkTest"
      name="checkTest"
      type="checkbox"
      value=""
    />
    <span
      class="css-qrnin0-CheckBoxSpan eynslpk3"
    />
  </label>
</div>
`;
